das/s3_storage_service.go

```diff
@@ -36,20 +46,20 @@ func NewS3StorageService(s3Config genericconf.S3Config) (StorageService, error)
 }
 
 func (s3s *S3StorageService) GetByHash(ctx context.Context, key []byte) ([]byte, error) {
-	var ret []byte
-	_, err := s3s.downloader.Download(ctx, manager.NewWriteAtBuffer(ret), &s3.GetObjectInput{
+	buf := manager.NewWriteAtBuffer([]byte{})
+	_, err := s3s.downloader.Download(ctx, buf, &s3.GetObjectInput{
 		Bucket: aws.String(s3s.s3Config.Bucket),
-		Key:    aws.String(base32.StdEncoding.EncodeToString(key)),
+		Key:    aws.String(string(key)),
```

Comment:
I think you actually want to encode the keys here:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html

Reply:
Done, thanks for catching this.